apps/consulting/services/posts/getAllCategories/utils/isValidCategoryItem.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  isValidObject,
+  isValidString,
+  trimWhiteSpace,
+  removeWhiteSpace,
+} from '@quansight/shared/utils';
+
+import { TRawCategoryItem } from '../types';
+
+const checkCategoryName = (name: string) =>
+  trimWhiteSpace(name).length === name.length ? name : false;
+
+const checkCategoryValue = (name: string) =>
+  removeWhiteSpace(name).length === name.length ? name : false;
+
+const cetegorySchema = {
+  name: checkCategoryName,
+  value: checkCategoryValue,
+};
+
+export const isValidCategoryItem = (categoryItem: TRawCategoryItem): boolean =>
```

Comment:
So when it comes to categories from Storyblok and local file. I first check if they both arrays. Then I check each individual object - and this is what `isValidCategoryItem`. So first, it should be and object `{}` with a `name` and `value` properties. Property `name` should be a string that sould not have any white space at the beginning or end but the one between words is allowed. So cases like ` Name`,  `Name `, ` Name   ` etc... are not allowed as names. Property `value` used for slugs, should not have any whitespace at all as we agree for posts slugs to not be like `my+sample+slug` but more `my-sample-slug`. So it checks for and white space at the beginning, end or middle of `value` to avoid cases like `sample -slug` which would probably reslove as `sampe+-slug` etc...

Reply:
Resolving as per the discussion in the main PR conversation thread.